libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -480,6 +480,15 @@ def build_extra(cls, values: Dict[str, Any]) -> Any:
         values = _build_model_kwargs(values, all_required_field_names)
         return values
 
+    @model_validator(mode="before")
+    @classmethod
+    def validate_temperature(cls, values: Dict[str, Any]) -> Any:
+        """Currently o1 models only allow temperature=1."""
+        model = values.get("model_name") or values.get("model") or ""
+        if "o1" in model and "temperature" not in values:
```

Comment:
```suggestion
        if model.startswith("o1-") and "temperature" not in values:
```

Reply:
also should we handle the 0.7 default here so it's less confusing pydantic validator/default wise, and remove default on attribute above?

Reply:
if we remove default or make it Optional might lead to linting errors for folks

Reply:
(re o1 vs o1- as prefix, presumably openai will soon release just an "o1" model)